File: 2275 Largest Combination With Bitwise AND Greater Than Zero/2275.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        bins = [0]*24

        for x in candidates:
            i = 0
            while x:
                if x % 2 == 1:
                    bins[i] += 1
                i += 1
                x //= 2
        
        return max(bins)

File: 2275 Largest Combination With Bitwise AND Greater Than Zero/2275.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        vector<int> bins(24, 0);

        for (int x: candidates){
            int i = 0;
            while (x){
                bins[i++] += x & 1;
                x /= 2;
            }
        }
        
        return *max_element(bins.begin(), bins.end());
    }
};
